src/omv/modules/py_tf.c

```diff
@@ -645,81 +891,112 @@ STATIC mp_obj_t py_tf_detect(uint n_args, const mp_obj_t *pos_args, mp_map_t *kw
+
+        if (!list_size(&thresholds)) {
+            color_thresholds_list_lnk_data_t lnk_data;
+            lnk_data.LMin = GRAYSCALE_MID;
+            lnk_data.LMax = GRAYSCALE_RANGE;
+            lnk_data.AMin = COLOR_A_MIN;
+            lnk_data.AMax = COLOR_A_MAX;
+            lnk_data.BMin = COLOR_B_MIN;
+            lnk_data.BMax = COLOR_B_MAX;
+            list_push_back(&thresholds, &lnk_data);
         }
 
-        out_list->items[i] = objects_list;
+        mp_obj_list_t *img_list = (mp_obj_list_t *) py_tf_segment_output_callback_data;
+        out_list = mp_obj_new_list(img_list->len, NULL);
+
+        float fscale = 1.f / GRAYSCALE_RANGE;
+        for (int i = 0, ii = img_list->len; i < ii; i++) {
+            image_t *img = py_image_cobj(img_list->items[i]);
+            float x_scale = roi.w / ((float) img->w);
+            float y_scale = roi.h / ((float) img->h);
+            // MAX == KeepAspectRatioByExpanding - MIN == KeepAspectRatio
+            float scale = IM_MIN(x_scale, y_scale);
+            int x_offset = fast_floorf((roi.w - (img->w * scale)) / 2.0f) + roi.x;
+            int y_offset = fast_floorf((roi.h - (img->h * scale)) / 2.0f) + roi.y;
+
+            list_t out;
+            imlib_find_blobs(&out, img, &((rectangle_t) {0, 0, img->w, img->h}), 1, 1,
+                             &thresholds, invert, 1, 1, false, 0,
+                             NULL, NULL, NULL, NULL, 0, 0);
+
+            size_t len = list_size(&out);
+            mp_obj_list_t *objects_list = mp_obj_new_list(len, NULL);
+            for (size_t j = 0; j < len; j++) {
+                find_blobs_list_lnk_data_t lnk_data;
+                list_pop_front(&out, &lnk_data);
+
+                histogram_t hist;
+                hist.LBinCount = GRAYSCALE_RANGE + 1;
+                hist.ABinCount = 0;
+                hist.BBinCount = 0;
+                hist.LBins = fb_alloc(hist.LBinCount * sizeof(float), FB_ALLOC_NO_HINT);
+                hist.ABins = NULL;
+                hist.BBins = NULL;
+                imlib_get_histogram(&hist, img, &lnk_data.rect, &thresholds, invert, NULL);
+
+                statistics_t stats;
+                imlib_get_statistics(&stats, img->pixfmt, &hist);
+                fb_free(); // fb_alloc(hist.LBinCount * sizeof(float), FB_ALLOC_NO_HINT);
+
+                py_tf_classification_obj_t *o = m_new_obj(py_tf_classification_obj_t);
+                o->base.type = &py_tf_classification_type;
+                o->x = mp_obj_new_int(fast_floorf(lnk_data.rect.x * scale) + x_offset);
+                o->y = mp_obj_new_int(fast_floorf(lnk_data.rect.y * scale) + y_offset);
+                o->w = mp_obj_new_int(fast_floorf(lnk_data.rect.w * scale));
+                o->h = mp_obj_new_int(fast_floorf(lnk_data.rect.h * scale));
+                o->output = mp_obj_new_float(stats.LMean * fscale);
+                objects_list->items[j] = o;
+            }
+
+            out_list->items[i] = objects_list;
+        }
```

Comment:
I think this code block should be moved to `py_tf_classify_output_callback` and the output constructed there.